### ISSUE TITLE
[pickers] New component: TimeRangePicker

### ISSUE THREAD

**Issue by ErnestBrandi** [mui/mui-x#4460]
## Benchmarks

- https://demo.mobiscroll.com/jquery/range/time-range#

**Comment by am-vishal** [comment]
we need time range picker in @mui something like this.
I found many people facing similar issues.
https://stackoverflow.com/questions/72811784/from-time-and-to-time-validation-in-react-dropdown-react-datepicker-hour

**Comment by simon-guider** [comment]
+1

**Comment by tfletch5** [comment]
+1

**Comment by Abdelrahman-Rabi** [comment]
+1

**Comment by a-ahandani** [comment]
+1

**Comment by WinmezzZ** [comment]
+1

**Comment by filipstojakovic** [comment]
+1

**Comment by davidsl4** [comment]
+1

**Comment by submetu** [comment]
+1

**Comment by bulatkhayrullin** [comment]
+1

**Comment by gcardacci** [comment]
+1

**Comment by NonozgYtb** [comment]
+1

**Comment by itseasy21** [comment]
+1

**Comment by flaviendelangle** [comment]
Small update here.
We know that a lot of people are waiting for this feature and it remains one of our top priority.

The team is currently rebuilding the input part of the pickers from scratch to provide a better experience.
We should be able to soon release a first documented (but still unstable) version of those components soon.
You can find a basic internal documentation of those components [here](https://next.mui.com/x/react-date-pickers/date-field) if you are curious.

The plan is then to create new keyboard-only components from *time range* and *date time range* in the coming weeks.
And then to build new pickers around them, just like we plan to migrate the already existing pickers to those new components.

In summary the `TimeRangePicker` will come in two steps:

1. It's keyboard-only component
2. It's full picker with the calendar views

**Comment by flaviendelangle** [comment]
As mentioned a few days ago, we now have a new component for date time range editing called `MultiInputTimeRangeField` (very long name, but a `SingleInputTimeRangeField` is in the pipeline).

The documentation is available [here](https://next.mui.com/x/react-date-pickers/time-range-field/)
It is still very early-stage but will serve as the baseline for the future `TimeRangePicker`.
So the more feedback we have to improve it, the easier it will be to build this new picker.

**Comment by mahouha** [comment]
+1

**Comment by DvirMalka** [comment]
+1

**Comment by dreemorra** [comment]
+1

**Comment by FlorianCassayre** [comment]
Guys, be aware that this feature will only be made available in the paid plan, not the "community" version.

**Comment by flaviendelangle** [comment]
Indeed :+1:

**Comment by IlliaHreben** [comment]
Exactly what I need!

**Comment by Rchrdlss3** [comment]
Would love to see this soon!

**Comment by svez01** [comment]
+1

**Comment by lobototoro** [comment]
+1

**Comment by robertdavidwest** [comment]
+1 ! <3

**Comment by ahmadgaz** [comment]
+1 !!

**Comment by AnvarAka1** [comment]
+1

**Comment by PFJoey** [comment]
+1

**Comment by rdori** [comment]
+1

**Comment by dabi-vm** [comment]
+1

**Comment by AmirHmZz** [comment]
+1

**Comment by Lucas-Heitele** [comment]
+1

**Comment by spmsh** [comment]
+1 thanks!

**Comment by danielcolgan** [comment]
+1

**Comment by kennie26** [comment]
+1

**Comment by dedegkaev** [comment]
+1

**Comment by charleneyt** [comment]
+1

**Comment by EitanElbaz** [comment]
+1

**Comment by JStevens127** [comment]
+1

**Comment by kushspatel** [comment]
+1

**Comment by Chao00** [comment]
+1

**Comment by big-duke** [comment]
+1

**Comment by Abusayid693** [comment]
+1

**Comment by nezerlee** [comment]
+1

**Comment by singagain** [comment]
+1

**Comment by iamtomhanks** [comment]
Yeeeeeee

**Comment by CJohnson25** [comment]
+1

**Comment by SergeiKachenia** [comment]
+1

**Comment by dave502** [comment]
+1

**Comment by Teese999** [comment]
+1

**Comment by wimpywarlord** [comment]
Very much needed

**Comment by BelindaPickering** [comment]
upvoted - yes please!

**Comment by mdzwink** [comment]
+1

**Comment by OyeHan** [comment]
+1

**Comment by armandobretas** [comment]
+1

**Comment by GutZuFusss** [comment]
+1

**Comment by lik-epg** [comment]
+1

**Comment by catx13** [comment]
+1

**Comment by onedotsix** [comment]
+1 Yes, please. I could use this right away...

**Comment by amillemanthree30** [comment]
+1 definitely could use immediately for custom scheduling application purposes

**Comment by georgeschulz** [comment]
+1 This would be great!

**Comment by HexaJohn** [comment]
+1 Would love to see this!

**Comment by mahatosourav91** [comment]
+1 Would love to see this!

**Comment by JoaoAMarinho** [comment]
+ 1

**Comment by rcasca** [comment]
+1

**Comment by amirnz026** [comment]
+1

**Comment by D2XHibinger** [comment]
+1

**Comment by HolyNoodle** [comment]
+1

**Comment by kpsolo** [comment]
+1

**Comment by abraham-shimelis** [comment]
Yes, it's needed

**Comment by amcmilne** [comment]
+1

**Comment by creightonfrance** [comment]
+1

**Comment by Annastrin** [comment]
+1

**Comment by ngouy** [comment]
@michaldudak could you put a hard number on ` if we see significant interest in the community, we may consider creating one.` please?

Some might consider 700+ as significant, some other not. Curious on the number you have in mind in these situations

**Comment by flaviendelangle** [comment]
We don't have fixed number because it depends a lot on the complexity of the feature.
A small feature might be created after just a dozen up-votes, a whole new set of component will of course require more traction.
For the `TimeRangePicker`, it's under development :+1:

**Comment by tcho-livex** [comment]
+1

**Comment by fnedic** [comment]
+1

**Comment by harrysingh** [comment]
+1

**Comment by Michael-Patrick-Lynch** [comment]
+1

**Comment by ingrid0b** [comment]
+1

**Comment by SaptooAji** [comment]
+1

**Comment by macmartins** [comment]
+1

**Comment by oliviertassinari** [comment]
In https://mui.com/blog/mui-x-v8-alpha-zero/ we communicated that MUI X v8.0 aims to include this issue.
